plugins/aws-appmesh/config/netconfig.go

```diff
@@ -40,8 +40,8 @@ type NetConfig struct {
 	EnableIPv6         bool
 }
 
-// netConfigJSON defines the network configuration JSON file format for the aws-appmesh cni plugin.
-type netConfigJSON struct {
+// NetConfigJSON defines the network configuration JSON file format for the aws-appmesh cni plugin.
+type NetConfigJSON struct {
```

Comment:
netConfigJSON is the internal representation of the JSON file, which is why it was private to the config package. Can we use the exported NetConfig structure in the e2etests instead? See my other comments below.

Reply:
IIUC it should be the other way. The input to CNI is json file and hence netconfigjson should be the interface (e.g. aws-appmesh.conf). e2e_test need to assert if the observed behavior against the json input. I could create a separate object to represent this json struct, but will end up creating netconfigjson again.